lib/vmdb/initializer.rb

```diff
@@ -15,11 +15,24 @@ def self.init
     end
 
     def self.init_secret_token
-      return if Rails.application.config.secret_key_base
+      # base will be generated from local file in a different format if nil
+      base = Rails.application.config.secret_key_base
+      db_token = session_secret_token
 
-      token = session_secret_token || SecureRandom.hex(64)
+      # skip if base is already set to db_token
+      return if base == db_token
 
-      Rails.application.config.secret_key_base = token
+      new_token = if db_token.present?
+                    # use db_token above all
+                    db_token
+                  else
+                    # generate a new token if base isn't already configured
+                    return if /^\h{128}$/.match?(base)
+
+                    SecureRandom.hex(64)
+                  end
```

Comment:
I don't understand this code relative to what was there before. It's like basically the same but not?  Can we discuss?